src/client/Microsoft.Identity.Client/TelemetryCore/TelemetryManager.cs

```diff
@@ -137,6 +147,93 @@ public void Flush(string correlationId)
             Callback?.Invoke(eventsToFlush.Cast<Dictionary<string, string>>().ToList());
         }
 
+        public string FetchAndResetPreviousHttpTelemetryContent()
+        {
+            if (!string.IsNullOrEmpty(_previousTelemetryPayload.ApiId))
+            {
+                string csv = CreateCSVForPreviousHttpTelem();
+                _currentTelemetryPayload.ResetLastErrorCode();
+                return csv;
+            }
+            else
+            {
+                return string.Empty;
+            }
+        }
+
+        public string FetchAndResetCurrentHttpTelemetryContent()
+        {
+            if (!string.IsNullOrEmpty(_currentTelemetryPayload.ApiId))
+            {               
+                string csv = CreateCSVForCurrentHttpTelem();
+                _currentTelemetryPayload.ResetLastErrorCode();
+                return csv;
+            }
+            else
+            {
+                return string.Empty;
+            }
+        }
+
+        private string CreateCSVForPreviousHttpTelem()
+        {
+            // csv expected format:
+            // 1|api_id, correlation_id, last_error_code
+            string[] myValues = new string[] {
+                _previousTelemetryPayload.ApiId,
+                _previousTelemetryPayload.CorrelationId,
+                _previousTelemetryPayload.LastErrorCode};
+
+            string csvString = string.Join(",", myValues);
+            csvString = TelemetryConstants.HttpTelemetrySchemaVersion1 + csvString;
+            return csvString;
+        }
+
+        private string CreateCSVForCurrentHttpTelem()
+        {
+            // csv expected format:
+            // 1|api_id|platform_config
```

Comment:
The actual fomat seems to be 1|api_id|correlation_id

Reply:
the format was right...i removed correlation_id

---
In reply to: [325638151](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/1389#discussion_r325638151) [](ancestors = 325638151)